### PR TITLE
DOC: clip()  allows arguments.

### PR DESCRIPTION
**Pull request by felixdivo**
This works in practice and could thus be reflected by the docs, right?

### CODE REVIEW

**Review by eric-wieser — COMMENTED** [review]
Looks reasonable. Maybe need to change "Array" on the next line too, but not a big deal.

Arguably `array_like` could be considered to include scalars, but the rest of this docstring doesn't seem to buy that argument.

**Comment by felixdivo** [comment]
Yeah, so there is more of this floating around. According to the [glossary](https://numpy.org/doc/stable/glossary.html#term-array-like), you are right:

> Any sequence that can be interpreted as an ndarray. This includes nested lists, tuples, scalars and existing arrays.

Even tough scalars are not really sequences, the show up in that glossary entry. What about updating the above glossary entry to "Any _object_ that can be interpreted as an ndarray."?

However, there seems to be more confusion around this, e.g. with [numpy.select](https://numpy.org/doc/stable/reference/generated/numpy.select.html). Here, lists of `scalar`s are accepted too, even though is only explicitly mentions `ndarray`s.

Maybe this should be discussed more widely?

**Comment by felixdivo** [comment]
I just noticed that #16996 might be a good place to discuss that.

**Comment by charris** [comment]
I think `array_like` should cover things, everything goes through arrays internally. I think it is the other parameter descriptions that need changing.

**Comment by bjnath** [comment]
Wording of latest #16996 commit (4dfdde4) is:

>Any scalar or sequence that can be interpreted as an ndarray. In addition to ndarrays and scalars this category includes lists (possibly nested and with different element types) and tuples. Any argument accepted by numpy.array is array_like.

**Comment by felixdivo** [comment]
@bjnath your changes should cover the wording of `array_like` in the glossary nicely

@charris I'll change this PR accordingly and change the description of `a_min` and `a_max`

**Comment by felixdivo** [comment]
Done.

**Review by rossbar — COMMENTED** [review]
If we're getting rid of the `scalar or`, then the "If `a_min` or `a_max` are array_like..." in the `a_max` parameter description should be reworded/removed.

Despite the fact that `array_like` technically includes scalars, I'm not sure that removing `scalar` here improves clarity re: how the function behaves. Not a blocker though, just a comment!

File: numpy/core/fromnumeric.py [hunk 1]
```diff
@@ -2059,11 +2059,11 @@ def clip(a, a_min, a_max, out=None, **kwargs):
...
-    a_min : scalar or array_like or None
+    a_min : array_like or None
         Minimum value. If None, clipping is not performed on lower
         interval edge. Not more than one of `a_min` and `a_max` may be
         None.
-    a_max : scalar or array_like or None
+    a_max : array_like or None
         Maximum value. If None, clipping is not performed on upper
         interval edge. Not more than one of `a_min` and `a_max` may be
         None. If `a_min` or `a_max` are array_like, then the three
```
Comment (eric-wieser) [inline numpy/core/fromnumeric.py:2069]: Perhaps
```
`a_min` and  `a_max` are broadcast against `a`.
```
in response to @rossbar's comment

**Review by rossbar — APPROVED** [review]
LGTM  thanks @felixdivo !

File: numpy/core/fromnumeric.py [hunk 2]
```diff
@@ -2059,15 +2059,14 @@ def clip(a, a_min, a_max, out=None, **kwargs):
...
         Minimum value. If None, clipping is not performed on lower
         interval edge. Not more than one of `a_min` and `a_max` may be
         None.
-    a_max : scalar or array_like or None
+    a_max : array_like or None
         Maximum value. If None, clipping is not performed on upper
         interval edge. Not more than one of `a_min` and `a_max` may be
-        None. If `a_min` or `a_max` are array_like, then the three
-        arrays will be broadcasted to match their shapes.
+        None. `a_min` and  `a_max` are broadcast against `a`.
```
Comment (eric-wieser) [inline numpy/core/fromnumeric.py:2069]: I wonder if it makes sense to combine these into a single docstring,
```
a_min, a_max : array_like or None
    Minimum and maximum value. If None, clipping is not performed on the corresponding edge. Not more than one of `a_min` and `a_max` may be None. These are broadcast against `a`.

Comment (felixdivo) [reply numpy/core/fromnumeric.py:2069]: That definitely looks better (as in easier to read). I also thought about that but was unsure whether that is OK formatting wise. I'll change that.

**Comment by mattip** [comment]
Thanks @felixdivo

**Pull request merged**